tests/validation/tests/v3_api/test_import_rke2_cluster.py

```diff
@@ -227,46 +265,66 @@ def create_rancher_cluster(client, rke2_clusterfilepath):
 
 
 def check_cluster_status(kubeconfig):
-    nodeNotReady = True
-    retries =0
+    print("Checking cluster status for {} server and {} agents nodes...".format(RANCHER_RKE2_NO_OF_SERVER_NODES, (int(RANCHER_RKE2_NO_OF_WORKER_NODES) + int(RANCHER_RKE2_NO_OF_WINDOWS_WORKER_NODES))))
+    retries = 0
+    actual_count_of_nodes = 0
+    expected_count_of_nodes = int(RANCHER_RKE2_NO_OF_SERVER_NODES) + \
+                              int(RANCHER_RKE2_NO_OF_WORKER_NODES) + \
+                              int(RANCHER_RKE2_NO_OF_WINDOWS_WORKER_NODES)
     try:
-        while nodeNotReady and (retries < 10):
-            cmd = "kubectl get nodes --no-headers -A --kubeconfig=" + kubeconfig
-            nodes = execute_command(cmd, False)
-            nodeNotReady = False
-            for node in nodes.strip().split("\n"):
-                state = node.split()[1]
-                if state != "Ready":
-                    nodeNotReady = True
-                if not nodeNotReady:
-                    break
+        # Retry logic for matching node count for 5 mins
```

Comment:
In your testing, were there any flakes based on the timings? I feel like we should also wait for 5 minutes on the node/pod Ready/Running statuses too instead of just 2 minutes there since RKE2 can be slow

Reply:
I have not seen any flakes based on timing. But I will take your suggestion to add 5 minutes to nodes and pods checking to be safe... totally agree on rke2 slowness :p